src/pokemon.c

```diff
@@ -8273,30 +8273,33 @@ u16 GetFormChangeTargetSpeciesBoxMon(struct BoxPokemon *boxMon, u16 method, u32
                         targetSpecies = formChanges[i].targetSpecies;
                     break;
                 case FORM_ITEM_USE:
-                    if (arg == formChanges[i].param1)
-                        targetSpecies = formChanges[i].targetSpecies;
-                    break;
-                case FORM_MOVE:
-                    if (BoxMonKnowsMove(boxMon, formChanges[i].param1) != formChanges[i].param2)
-                        targetSpecies = formChanges[i].targetSpecies;
-                    break;
-                case FORM_ITEM_USE_TIME:
                     RtcCalcLocalTime();
                     if (arg == formChanges[i].param1)
                     {
-                        switch (formChanges[i].param2)
+                        if (!formChanges[i].param2)
```

Comment:
There's no need for this check. If param2 is 0 (which it is by default), no case on the switch will trigger and the targetSpecies will not change.